src/blob.jl

```diff
@@ -61,11 +61,28 @@ function Base.:-(blob1::Blob, blob2::Blob)
     getfield(blob1, :offset) - getfield(blob2, :offset)
 end
 
-@noinline function _throw_assert_not_null_error(typename::Symbol)
-    throw(AssertionError("Null pointer dereference in Blob{$(typename)}"))
+@eval @inline BOUNDSCHECK_ON_DEREF_ENABLED() = false
+
+function enable_boundscheck_on_deref()
```

Comment:
do we need this to be enabled/disabled at runtime? If not, we could have it be a compile-time option e.g. with Preferences.jl e.g. 
```julia
const BOUNDSCHECK_ON_DEREF_ENABLED::Bool = get_preference("BLOBS_BOUNDSCHECK_ON_DEREF_ENABLED", false)
```